django/test/testcases.py

```diff
@@ -306,10 +307,13 @@ def assertRedirects(self, response, expected_url, status_code=302,
                 # netloc might be empty, or in cases where Django tests the
                 # HTTP scheme, the convention is for netloc to be 'testserver'.
                 # Trust both as "internal" URLs here.
-                if netloc and netloc != 'testserver':
+                domain, port = split_domain_port(netloc)
+                if domain and not validate_host(domain, settings.ALLOWED_HOSTS):
                     raise ValueError(
                         "The Django test client is unable to fetch remote URLs (got %s). "
-                        "Use assertRedirects(..., fetch_redirect_response=False) instead." % url
+                        "If the host should be served by Django, you may need to add %s "
```

Comment:
I'm not sure about "you may need to" -- do you have a case in mind where it isn't the case?